service/controller/clusternamespace/resource/rbacappoperator/config.go

```diff
@@ -0,0 +1,203 @@
+				Verbs:     []string{"list", "watch"},
+			},
+		},
+	}
+}
+
+func getAppOperatorCLusterRoleBinding(ns corev1.Namespace, clusterRoleRef *rbacv1.ClusterRole) *rbacv1.ClusterRoleBinding {
+	return &rbacv1.ClusterRoleBinding{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ClusterRole",
+			APIVersion: "rbac.authorization.k8s.io/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: key.AppOperatorRbacOperatorManagedResourceName(ns),
+			Labels: map[string]string{
+				label.ManagedBy: project.Name(),
+			},
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      key.AppOperatorServiceAccountNameFromNamespace(ns),
+				Namespace: ns.Name,
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     clusterRoleRef.Kind,
+			Name:     clusterRoleRef.Name,
+		},
+	}
+}
+
+func getAppOperatorCatalogReaderRole(ns corev1.Namespace) *rbacv1.Role {
+	return &rbacv1.Role{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Role",
+			APIVersion: "rbac.authorization.k8s.io/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: key.AppOperatorRbacOperatorManagedResourceName(ns),
+			Labels: map[string]string{
+				label.ManagedBy: project.Name(),
+			},
+			Namespace: "giantswarm",
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{""},
+				Resources: []string{"configmaps"},
+				Verbs:     []string{"get", "patch"},
+			},
+		},
+	}
+}
+
+func getAppOperatorCatalogReaderRoleBinding(ns corev1.Namespace, roleRef *rbacv1.Role) *rbacv1.RoleBinding {
+	return &rbacv1.RoleBinding{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "RoleBinding",
+			APIVersion: "rbac.authorization.k8s.io/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: key.AppOperatorRbacOperatorManagedResourceName(ns),
+			Labels: map[string]string{
+				label.ManagedBy: project.Name(),
+			},
+			Namespace: "giantswarm",
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      key.AppOperatorServiceAccountNameFromNamespace(ns),
+				Namespace: ns.Name,
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     roleRef.Kind,
+			Name:     roleRef.Name,
+		},
+	}
+}
+
+func getAppOperatorOwnNamespaceRole(ns corev1.Namespace) *rbacv1.Role {
+	return &rbacv1.Role{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Role",
+			APIVersion: "rbac.authorization.k8s.io/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: key.AppOperatorRbacOperatorManagedResourceName(ns),
+			Labels: map[string]string{
+				label.ManagedBy: project.Name(),
+			},
+			Namespace: ns.Name,
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{""},
+				Resources: []string{"namespaces"},
+				Verbs:     []string{"get"},
+			},
+			{
+				APIGroups: []string{""},
+				Resources: []string{"configmaps"},
+				Verbs:     []string{"*"},
+			},
+			{
+				APIGroups: []string{""},
+				Resources: []string{"secrets"},
+				Verbs:     []string{"*"},
+			},
```

Comment:
The original role had some permissions for Events and app-operators still emit them in their own namespace.

```suggestion
			},
			{
				APIGroups: []string{""},
				Resources: []string{"events"},
				Verbs:     []string{"create", "patch", "update"},
			},
```